### PR TITLE
Add logging to the default evaluator

### PR DESCRIPTION
**Pull request by sawagh**
This surfaces the rejected matches, with information about what matches did what ticket collide with scores of both matches. This helps debug, correlate rejected matches with match function runs.

### CODE REVIEW

File: examples/evaluator/golang/simple/evaluate/evaluator.go [hunk 1]
```diff
@@ -79,20 +86,35 @@ func Evaluate(p *harness.EvaluatorParams) ([]*pb.Match, error) {
...
 	for _, t := range m.match.GetTickets() {
-		if _, ok := d.ticketsUsed[t.Id]; ok {
+		if cm, ok := d.ticketsUsed[t.Id]; ok {
+			logger.WithFields(logrus.Fields{
+				"match_id":              m.match.GetMatchId(),
+				"ticket_id":             t.GetId(),
+				"match_score":           m.inp.GetScore(),
+				"colliding_match_id":    cm.id,
+				"colliding_match_score": cm.score,
+			}).Info("Higher quality match with colliding ticket found. Rejecting match.")
```
Comment (yfei1) [inline examples/evaluator/golang/simple/evaluate/evaluator.go]: Maybe `logger.Debug`?

Comment (sawagh) [reply examples/evaluator/golang/simple/evaluate/evaluator.go]: lets keep a higher tracing level for now specially for things that evaluator drops. I would prefer the demo executions to show this log statement by default

**Pull request merged**